sdk/storage/azure-storage-file-datalake/samples/datalake_samples_access_control_recursive_async.py

```diff
@@ -0,0 +1,113 @@
+    directory_client = await filesystem_client.create_directory(dir_name)
+
+    # populate the directory with some child files
+    await create_child_files(directory_client, 35)
+
+    # get and display the permissions of the parent directory
+    acl_props = await directory_client.get_access_control()
+    print("Permissions of directory '{}' are {}.".format(dir_name, acl_props['permissions']))
+
+    # set the permissions of the entire directory tree recursively
+    # update/remove acl operations are performed the same way
+    acl = 'user::rwx,group::r-x,other::rwx'
+    failed_entries = []
+
+    # the progress callback is invoked each time a batch is completed
+    async def progress_callback(resp):
+        print("In this batch: {} directories and {} files were processed successfully, {} failures were counted"
+              .format(resp.directories_successful, resp.files_successful, resp.failure_count))
+
+        # keep track of failed entries if there are any
+        failed_entries.append(resp.failed_entries)
+
+    # illustrate the operation by using a small batch_size
+    summary = await directory_client.set_access_control_recursive(acl=acl, progress_callback=progress_callback,
+                                                                  batch_size=5)
+    print("Summary: {} directories and {} files were updated successfully, {} failures were counted."
+          .format(summary.directories_successful, summary.files_successful, summary.failure_count))
+
+    # if an error was encountered, a continuation token would be returned if the operation can be resumed
+    if summary.continuation is not None:
+        print("The operation can be resumed by passing the continuation token {} again into the access control method."
+              .format(summary.continuation_token))
+
+    # get and display the permissions of the parent directory again
+    acl_props = await directory_client.get_access_control()
+    print("New permissions of directory '{}' and its children are {}.".format(dir_name, acl_props['permissions']))
+
+
+async def create_child_files(directory_client, num_child_files):
+    import itertools
+
+    async def create_file():
+        # generate a random name
+        file_name = str(uuid.uuid4()).replace('-', '')
+        file_client = directory_client.get_file_client(file_name)
+        await file_client.create_file()
+
+    futures = [asyncio.ensure_future(create_file()) for _ in itertools.repeat(None, num_child_files)]
+    await asyncio.wait(futures)
+    print("Created {} files under the directory '{}'.".format(num_child_files, directory_client.path_name))
+
+
+async def run():
+    account_name = os.getenv('STORAGE_ACCOUNT_NAME', "")
+    account_key = os.getenv('STORAGE_ACCOUNT_KEY', "")
+
+    # set up the service client with the credentials from the environment variables
+    service_client = DataLakeServiceClient(account_url="{}://{}.dfs.core.windows.net".format(
```

Comment:
Did you see any warning message when you ran the sample?
I'm just curious why no context manager was used and there's no warning thrown.
When I added async samples I had to use `async with service_client:` to call the __ aenter __ and __ aexit ___ automatically else there were some unclosed connection warning/error.

Reply:
Good catch, I didn't notice the warning 😺